packages/rum-core/src/boot/rumPublicApi.ts

```diff
@@ -152,21 +154,25 @@ export function makeRumPublicApi<C extends RumUserConfiguration>(startRumImpl: S
       rumPublicApi.addAction(name, context as Context)
     },
 
-    addError: monitor((error: unknown, context?: object, source: ProvidedSource = ErrorSource.CUSTOM) => {
-      let checkedSource: ProvidedSource
-      if (source === ErrorSource.CUSTOM || source === ErrorSource.NETWORK || source === ErrorSource.SOURCE) {
-        checkedSource = source
-      } else {
-        display.error(`DD_RUM.addError: Invalid source '${source as string}'`)
-        checkedSource = ErrorSource.CUSTOM
-      }
-      addErrorStrategy({
-        error,
-        context: deepClone(context as Context),
-        source: checkedSource,
-        startClocks: clocksNow(),
+    addError: (error: unknown, context?: object, source: ProvidedSource = ErrorSource.CUSTOM) => {
+      const instrumentationError = createInstrumentationError()
```

Comment:
I wanted to put as less domain logic as possible here. 
`createInstrumentationError` is only here to ensure IE compatibility. 
Could also be a util function called `createErrorWithCallstack`